test/tests/functional/pbs_nonprint_characters.py

```diff
@@ -947,18 +944,22 @@ def test_terminal_control_hook_name(self):
         """
         # Create hook
         hook_name = "h%s%sd" % (self.bold, self.red)
-        a = {'event': "execjob_begin", 'enabled': 'True', 'debug': 'True'}
-        try:
-            rv = self.server.create_hook(hook_name, a)
-        except PbsManagerError:
+        cmd = ['qmgr', '-c', 'create hook %s' % hook_name]
+        ret = self.du.run_cmd(self.server.hostname, cmd, sudo=True)
+        if ret['rc'] != 0:
             # Delete the hook first then create the hook
-            self.server.manager(MGR_CMD_DELETE, HOOK, id=hook_name)
-            rv = self.server.create_hook(hook_name, a)
-        self.assertTrue(rv)
+            cmd = ['qmgr', '-c', 'delete hook %s' % hook_name]
```

Comment:
I don't understand. why would the hook creation fail at the first time?

Reply:
If a hook with the same name already exists then hook creation will fail.

Reply:
agreed. So you want to avoid using same names for hooks in the same test or move this to the setUp (deletion of hook part in case PTL setUp has failed or hook has not properly cleaned up by mom for some reason).

Reply:
I have the delete hook there in the very remote chance that the same hook name already existed.  The tests use different names for the hooks.

Reply:
So I looked at the whole test and I still don't like the way this is done. how about checking if the hook_name exists, if yes then only delete it else go ahead and create them.

Reply:
Done